src/limb_repo/approaches/lr_system.py

```diff
@@ -0,0 +1,26 @@
+"""A base class defining the API for a Limb Repositioning System."""
+
+import abc
+
+from limb_repo.environments.base_env import BaseEnv
+from limb_repo.structs import Controller, State
+
+
+class LimbRepositioningSystem:
+    """A base class defining the API for a Limb Repositioning System."""
+
+    def __init__(
+        self,
+        environment: BaseEnv,
+        controller: Controller,
+    ) -> None:
+        self.environment: BaseEnv = environment
+        self.controller: Controller = controller
+
+    @abc.abstractmethod
+    def send_control(self, state: State) -> None:
```

Comment:
looking at this in isolation, I find it quite confusing -- (1) what is an "environment agent"? (2) why does this method take in a state?